src/hashtable.c

```diff
@@ -979,6 +986,24 @@ static void prefetchNextBucketEntries(iter *iter, bucket *current_bucket) {
     }
 }
 
+/* Prefetches the values associated with filled entries in the given bucket. */
+static void prefetchBucketValues(bucket *b, hashtable *ht) {
+    if (!b->presence) return;
+    for (int pos = 0; pos < numBucketPositions(b); pos++) {
+        if (isPositionFilled(b, pos)) {
+            entryPrefetchValue(ht, b->entries[pos]);
+        }
+    }
```

Comment:
I guess it is easier to read without the entryPrefetchValue helper function and we can return early if there is no entryPrefetchValue callback. Then we can delete the entryPrefetchValue helper. WDYT?

```suggestion
    if (!b->presence || ht->type->entryPrefetchValue == NULL) return;
    for (int pos = 0; pos < numBucketPositions(b); pos++) {
        if (isPositionFilled(b, pos)) {
            ht->type->entryPrefetchValue(b->entries[pos]);
        }
    }
```

Reply:
I agree.